packages/ckeditor5-emoji/src/emojiutils.ts

```diff
@@ -44,6 +44,9 @@ const EMOJI_SUPPORT_LEVEL: Record<string, number> = {
 
 const BASELINE_EMOJI_WIDTH = 24;
 
+const CANVAS_FOR_TESTING_EMOJI_WIDTH = /** #__PURE__ */ ( () => document.createElement( 'canvas' ).getContext( '2d' )! )();
+const CONTAINER_STYLES_MAP = /* #__PURE__ */ new Map<HTMLDivElement, CSSStyleDeclaration>();
```

Comment:
Consider moving both of the constants to a plugin class as private members. This way, accessing the `document` global will crash only at run-time instead of the import phase when running SSR.

The initialization can be done in the `Plugin#init()` method. Do we need to clean anything? Perhaps, on `Plugin#destroy()` we could leave a blank canvas to avoid leaking _what it does_.